src/lex.l

```diff
@@ -107,5 +107,4 @@ userdebug_or_eng { return USERDEBUG_OR_ENG; }
 (\#.*)?\n { yylineno++; }
 dnl(.*)?\n { yylineno++; }
 [ \t] ; /* normally skip whitespace */
-\% { yyerror("Invalid character: %%"); /* will be run through printf again */ }
```

Comment:
This was added because yyerror expects format strings, so "Invalid character: %", which is what the generic case would return would cause an error on printf in display_check_result.